src/sql/workbench/parts/editData/browser/editData.component.ts

```diff
@@ -408,8 +409,12 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
 	/**
 	 * Replace the line breaks with space.
 	 */
-	private spacefyLinebreaks(inputStr: string): string {
-		return inputStr.replace(/(\r\n|\n|\r)/g, ' ');
+	private replaceLinebreaks(inputStr: string): string {
+		let newlineMatches = inputStr.match(/(\r\n|\n|\r)/g);
+		if (newlineMatches && newlineMatches.length > 0) {
+			this.newlinePattern = newlineMatches[0];
```

Comment:
The content might have multiple line break types contained within it - I think you're going to have to store each one and then go back through and replace them one by one. Getting super picky here but it's pretty important to preserve the data exactly as it was so I think the extra bit of processing is worth the benefits.

Reply:
Content cannot have multiple type line breaks. That's OS specific, it would be different on different platforms but it will not have multiple types on a platform.

Reply:
Sure you can - line breaks are just special characters (0A and 0D specifically). Is it likely to happen? No, not really. But it could happen and we should do our best to ensure we're preserving the user content exactly as it was. 

![image](https://user-images.githubusercontent.com/28519865/63278129-8f176b80-c25b-11e9-9331-608498271dda.png)